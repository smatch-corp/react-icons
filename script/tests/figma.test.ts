import { expect, it } from 'vitest';

import { FigmaComponent, figma } from '../figma.js';

const FILE_ID = 'FILE_ID';
const NODE_ID = '1826:9210';

it('피그마 파일에서 컴포넌트 정보를 가져올 수 있다.', async () => {
    const components = await figma.getIconComponents(FILE_ID, NODE_ID);

    expect(components).toMatchSnapshot();
    expect(components.length).toMatchInlineSnapshot('166');
});

it('컴포넌트 정보를 활용해서 SVG 이미지를 가져올 수 있다.', async () => {
    const components: FigmaComponent[] = [
        { name: 'ArrowRight', node_id: '1:1' },
        { name: 'ArrowLeft', node_id: '1:2' },
    ];

    const images = await figma.getSvgImages(FILE_ID, components);

    expect(images).toMatchInlineSnapshot(`
      [
        {
          "component": {
            "name": "ArrowRight",
            "node_id": "1:1",
          },
          "imageUrl": "https://image.com/1:1",
        },
        {
          "component": {
            "name": "ArrowLeft",
            "node_id": "1:2",
          },
          "imageUrl": "https://image.com/1:2",
        },
      ]
    `);
});

it('컴포넌트 이름을 Icon 으로 시작하는 형태로 노멀라이즈 할 수 있다.', () => {
    expect(figma.normalizeComponentName('Icon/ArrowRight')).toBe('IconArrowRight');
    expect(figma.normalizeComponentName('IconArrowLeft')).toBe('IconArrowLeft');
    expect(figma.normalizeComponentName('ArrowUp')).toBe('IconArrowUp');
});

it('SVG가 포함된 컴포넌트 정보를 리액트 컴포넌트로 변환할 수 있다.', async () => {
    const result = await figma.convertToReactComponent({
        component: {
            name: 'ArrowLeft',
            node_id: '1:1',
        },
        imageUrl: 'https://image.com/1:1',
    });

    expect(result.code.includes('</svg>')).toBeTruthy();
    expect(result.code).toMatchInlineSnapshot(`
      "import type { SVGProps } from \\"react\\";
      import { Ref, forwardRef } from \\"react\\";
      const IconArrowLeft = (props: SVGProps<SVGSVGElement>, ref: Ref<SVGSVGElement>) => <svg width=\\"1em\\" height=\\"1em\\" viewBox=\\"0 0 24 24\\" fill=\\"none\\" xmlns=\\"http://www.w3.org/2000/svg\\" ref={ref} {...props}><path d=\\"M5 22C4.45 22 3.97917 21.8042 3.5875 21.4125C3.19583 21.0208 3 20.55 3 20V6C3 5.45 3.19583 4.97917 3.5875 4.5875C3.97917 4.19583 4.45 4 5 4H6V3C6 2.71667 6.09583 2.47917 6.2875 2.2875C6.47917 2.09583 6.71667 2 7 2C7.28333 2 7.52083 2.09583 7.7125 2.2875C7.90417 2.47917 8 2.71667 8 3V4H16V3C16 2.71667 16.0958 2.47917 16.2875 2.2875C16.4792 2.09583 16.7167 2 17 2C17.2833 2 17.5208 2.09583 17.7125 2.2875C17.9042 2.47917 18 2.71667 18 3V4H19C19.55 4 20.0208 4.19583 20.4125 4.5875C20.8042 4.97917 21 5.45 21 6V20C21 20.55 20.8042 21.0208 20.4125 21.4125C20.0208 21.8042 19.55 22 19 22H5ZM5 20H19V10H5V20ZM5 8H19V6H5V8Z\\" fill=\\"currentColor\\" /></svg>;
      const ForwardRef = forwardRef(IconArrowLeft);
      export { ForwardRef as IconArrowLeft };"
    `);
});
