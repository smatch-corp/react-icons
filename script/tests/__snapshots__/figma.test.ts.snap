// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`피그마 파일에서 컴포넌트 정보를 가져올 수 있다. 1`] = `
[
  {
    "name": "ManageAccounts",
    "node_id": "4574:2850",
  },
  {
    "name": "ManageAccountsFill",
    "node_id": "4574:2854",
  },
  {
    "name": "EditFill",
    "node_id": "4385:2272",
  },
  {
    "name": "ScheduleSendFill",
    "node_id": "4675:628",
  },
  {
    "name": "Cached",
    "node_id": "4680:645",
  },
  {
    "name": "ListEmpty",
    "node_id": "4229:32377",
  },
  {
    "name": "InventoryRounded",
    "node_id": "4229:31964",
  },
  {
    "name": "CheckCircleOutlineRounded",
    "node_id": "4229:31954",
  },
  {
    "name": "NotListedLocationRounded",
    "node_id": "4229:31961",
  },
  {
    "name": "RadioBlankDesigned",
    "node_id": "4229:32380",
  },
  {
    "name": "LocationPin",
    "node_id": "4229:32376",
  },
  {
    "name": "StreetView",
    "node_id": "4229:32375",
  },
  {
    "name": "RadioDesigned",
    "node_id": "4229:32373",
  },
  {
    "name": "LMComment",
    "node_id": "4229:32371",
  },
  {
    "name": "IndeterminateCheckBoxRoundedDesigned",
    "node_id": "4229:32370",
  },
  {
    "name": "ListFilter",
    "node_id": "4229:32368",
  },
  {
    "name": "CheckBoxRoundedDesigned",
    "node_id": "4229:32367",
  },
  {
    "name": "PDFFile",
    "node_id": "4229:32362",
  },
  {
    "name": "MapFilter",
    "node_id": "4229:32361",
  },
  {
    "name": "ListSearch",
    "node_id": "4229:32358",
  },
  {
    "name": "DistanceMeter",
    "node_id": "4229:32357",
  },
  {
    "name": "ClickButton",
    "node_id": "4229:32356",
  },
  {
    "name": "ListEdit",
    "node_id": "4229:32355",
  },
  {
    "name": "EventRounded",
    "node_id": "4229:32048",
  },
  {
    "name": "WheretoVoteRounded",
    "node_id": "4229:32044",
  },
  {
    "name": "ImageNotSupportedRounded",
    "node_id": "4229:32022",
  },
  {
    "name": "LocationOffRounded",
    "node_id": "4229:32030",
  },
  {
    "name": "MoreVertRounded",
    "node_id": "4229:32012",
  },
  {
    "name": "KeyboardArrowDownRounded",
    "node_id": "4229:32002",
  },
  {
    "name": "CheckRounded",
    "node_id": "4229:32006",
  },
  {
    "name": "AddRounded",
    "node_id": "4229:31989",
  },
  {
    "name": "AddCircleOutlineRounded",
    "node_id": "4229:31987",
  },
  {
    "name": "RemoveCircleRounded",
    "node_id": "4229:31980",
  },
  {
    "name": "SentimentSatisfiedRounded",
    "node_id": "4229:31974",
  },
  {
    "name": "LocationCityRounded",
    "node_id": "4229:31968",
  },
  {
    "name": "LocalFireDepartmentRounded",
    "node_id": "4229:31963",
  },
  {
    "name": "DateRangeRounded",
    "node_id": "4229:31967",
  },
  {
    "name": "AddPhotoAlternateRounded",
    "node_id": "4229:31946",
  },
  {
    "name": "SouthWestRounded",
    "node_id": "4229:31939",
  },
  {
    "name": "NorthEastRounded",
    "node_id": "4229:31938",
  },
  {
    "name": "SouthEastRounded",
    "node_id": "4229:31936",
  },
  {
    "name": "SortRounded",
    "node_id": "4229:31941",
  },
  {
    "name": "MenuRounded",
    "node_id": "4229:31998",
  },
  {
    "name": "WarningFillRounded",
    "node_id": "4229:31953",
  },
  {
    "name": "BoltRounded",
    "node_id": "4229:31951",
  },
  {
    "name": "KeyboardArrowRightRounded",
    "node_id": "4229:32001",
  },
  {
    "name": "ContentPasteRounded",
    "node_id": "4229:32029",
  },
  {
    "name": "KeyboardDoubleArrowRightRounded",
    "node_id": "4229:32025",
  },
  {
    "name": "ListChecked",
    "node_id": "4229:32359",
  },
  {
    "name": "ZoomOutRounded",
    "node_id": "4229:31942",
  },
  {
    "name": "NorthWestRounded",
    "node_id": "4229:31937",
  },
  {
    "name": "ApartmentRounded",
    "node_id": "4229:31969",
  },
  {
    "name": "ContentCopyRounded",
    "node_id": "4229:31947",
  },
  {
    "name": "EventNoteRounded",
    "node_id": "4229:32046",
  },
  {
    "name": "RemoveCircleOutlineRounded",
    "node_id": "4229:31991",
  },
  {
    "name": "RemoveRounded",
    "node_id": "4229:31978",
  },
  {
    "name": "CancelRounded",
    "node_id": "4229:32034",
  },
  {
    "name": "ContentPasteGoRounded",
    "node_id": "4229:32031",
  },
  {
    "name": "PaymentRounded",
    "node_id": "4229:31995",
  },
  {
    "name": "CloseRounded",
    "node_id": "4229:32015",
  },
  {
    "name": "PersonFill",
    "node_id": "4229:32027",
  },
  {
    "name": "NewShort",
    "node_id": "4229:32398",
  },
  {
    "name": "LoginRounded",
    "node_id": "4229:32005",
  },
  {
    "name": "LoadingSpinner",
    "node_id": "4229:32354",
  },
  {
    "name": "EmptyCheckList",
    "node_id": "4229:32360",
  },
  {
    "name": "MailCircleRounded",
    "node_id": "4229:32379",
  },
  {
    "name": "NotificationImportantRounded",
    "node_id": "4229:31983",
  },
  {
    "name": "SmatchCorp",
    "node_id": "4229:32387",
  },
  {
    "name": "KeyboardArrowLeftRounded",
    "node_id": "4229:32023",
  },
  {
    "name": "CheckBoxOutlineBlankRoundedDesigned",
    "node_id": "4229:32382",
  },
  {
    "name": "ErrorRounded",
    "node_id": "4229:32007",
  },
  {
    "name": "NewLMComment",
    "node_id": "4229:32372",
  },
  {
    "name": "Salesforce",
    "node_id": "4229:32374",
  },
  {
    "name": "ArrowBackRounded",
    "node_id": "4229:32043",
  },
  {
    "name": "CalendarToday",
    "node_id": "4229:31935",
  },
  {
    "name": "MailRounded",
    "node_id": "4229:32019",
  },
  {
    "name": "KeyboardArrowUpRounded",
    "node_id": "4229:32039",
  },
  {
    "name": "RuleRounded",
    "node_id": "4229:31952",
  },
  {
    "name": "ControlPointDuplicateRounded",
    "node_id": "4229:32417",
  },
  {
    "name": "GroupsFill",
    "node_id": "4227:30723",
  },
  {
    "name": "EditRounded",
    "node_id": "4229:31943",
  },
  {
    "name": "OpeninNewRounded",
    "node_id": "4229:32017",
  },
  {
    "name": "Won",
    "node_id": "4229:32381",
  },
  {
    "name": "HistoryRounded",
    "node_id": "4229:32433",
  },
  {
    "name": "Process",
    "node_id": "4229:32383",
  },
  {
    "name": "WonCircle",
    "node_id": "4229:32378",
  },
  {
    "name": "ProposalArrived",
    "node_id": "4229:32366",
  },
  {
    "name": "QuestionOutlineRounded",
    "node_id": "4229:32369",
  },
  {
    "name": "NaverBlog",
    "node_id": "4229:32384",
  },
  {
    "name": "ProposalList",
    "node_id": "4229:32363",
  },
  {
    "name": "InfoCircle",
    "node_id": "4229:32364",
  },
  {
    "name": "ProposalQuick",
    "node_id": "4229:32386",
  },
  {
    "name": "ProposalAdded",
    "node_id": "4229:32365",
  },
  {
    "name": "UpdateRounded",
    "node_id": "4229:32420",
  },
  {
    "name": "DeleteRounded",
    "node_id": "4229:32436",
  },
  {
    "name": "FileDownloadRounded",
    "node_id": "4229:32008",
  },
  {
    "name": "ShareRounded",
    "node_id": "4229:32018",
  },
  {
    "name": "SendRounded",
    "node_id": "4229:31945",
  },
  {
    "name": "SendFillRounded",
    "node_id": "4229:32454",
  },
  {
    "name": "LockRounded",
    "node_id": "4229:32457",
  },
  {
    "name": "BookmarkRounded",
    "node_id": "4229:32463",
  },
  {
    "name": "BookmarkFillRounded",
    "node_id": "4229:32462",
  },
  {
    "name": "CheckCircleRounded",
    "node_id": "4229:32037",
  },
  {
    "name": "LinkRounded",
    "node_id": "4229:32473",
  },
  {
    "name": "PublishedWithChangesRounded",
    "node_id": "4229:32492",
  },
  {
    "name": "CorporateFareRounded",
    "node_id": "4229:31966",
  },
  {
    "name": "AddCircleRounded",
    "node_id": "4229:31982",
  },
  {
    "name": "RefreshRounded",
    "node_id": "4229:32503",
  },
  {
    "name": "PhonelinkEraseRounded",
    "node_id": "4238:32519",
  },
  {
    "name": "TodayRounded",
    "node_id": "4238:32546",
  },
  {
    "name": "EventAvailableRounded",
    "node_id": "4229:32041",
  },
  {
    "name": "EventRepeatRounded",
    "node_id": "4229:32045",
  },
  {
    "name": "EventBusyRounded",
    "node_id": "4229:32042",
  },
  {
    "name": "AddAPhotoFillRounded",
    "node_id": "4238:32549",
  },
  {
    "name": "UploadRounded",
    "node_id": "4238:32558",
  },
  {
    "name": "SwapVertRounded",
    "node_id": "4238:32561",
  },
  {
    "name": "FilterAltRounded",
    "node_id": "4229:32009",
  },
  {
    "name": "SearchRounded",
    "node_id": "4238:32568",
  },
  {
    "name": "DirectionsSubwayFilledRounded",
    "node_id": "4229:31972",
  },
  {
    "name": "BorderColorRounded",
    "node_id": "4229:31984",
  },
  {
    "name": "MyLocationRounded",
    "node_id": "4229:32021",
  },
  {
    "name": "ErrorOutlineRounded",
    "node_id": "4229:32024",
  },
  {
    "name": "InfoRounded",
    "node_id": "4229:32033",
  },
  {
    "name": "AddLocationRounded",
    "node_id": "4229:32036",
  },
  {
    "name": "LocationOnRounded",
    "node_id": "4229:32047",
  },
  {
    "name": "Group",
    "node_id": "4376:38078",
  },
  {
    "name": "NotificationsRounded",
    "node_id": "4229:32035",
  },
  {
    "name": "Undo",
    "node_id": "4380:1126",
  },
  {
    "name": "DashboardCustomizeFill",
    "node_id": "4376:39118",
  },
  {
    "name": "Mail",
    "node_id": "4376:37836",
  },
  {
    "name": "VerifiedUserRounded",
    "node_id": "4229:32013",
  },
  {
    "name": "ThumbUpRounded",
    "node_id": "4229:32026",
  },
  {
    "name": "PictureasPdfRounded",
    "node_id": "4229:31965",
  },
  {
    "name": "PersonSearchRounded",
    "node_id": "4229:31985",
  },
  {
    "name": "QuestionRounded",
    "node_id": "4229:32385",
  },
  {
    "name": "StarRounded",
    "node_id": "4229:31992",
  },
  {
    "name": "KeyboardDoubleArrowLeftRounded",
    "node_id": "4229:31990",
  },
  {
    "name": "ConstructionRounded",
    "node_id": "4229:31956",
  },
  {
    "name": "BalanceRounded",
    "node_id": "4229:31981",
  },
  {
    "name": "CircleNotificationsRounded",
    "node_id": "4229:32020",
  },
  {
    "name": "ArrowForwardRounded",
    "node_id": "4229:32011",
  },
  {
    "name": "ArrowUpwardRounded",
    "node_id": "4229:32004",
  },
  {
    "name": "NotificationsActiveRounded",
    "node_id": "4229:31988",
  },
  {
    "name": "StarsRounded",
    "node_id": "4229:32032",
  },
  {
    "name": "VerifiedRounded",
    "node_id": "4229:32010",
  },
  {
    "name": "ZoomInRounded",
    "node_id": "4229:32016",
  },
  {
    "name": "RestartAltRounded",
    "node_id": "4229:31976",
  },
  {
    "name": "FavoriteBorderRounded",
    "node_id": "4229:32000",
  },
  {
    "name": "ThumbDownRounded",
    "node_id": "4229:31994",
  },
  {
    "name": "MapRounded",
    "node_id": "4229:31940",
  },
  {
    "name": "LogoutRounded",
    "node_id": "4229:31949",
  },
  {
    "name": "FormatLineSpacingRounded",
    "node_id": "4229:31950",
  },
  {
    "name": "FavoriteRounded",
    "node_id": "4229:31997",
  },
  {
    "name": "AssignmentIndRounded",
    "node_id": "4229:32040",
  },
  {
    "name": "ArrowDownwardRounded",
    "node_id": "4229:32028",
  },
  {
    "name": "StarBorderRounded",
    "node_id": "4229:31993",
  },
  {
    "name": "NotificationsOffRounded",
    "node_id": "4229:31957",
  },
  {
    "name": "HandShakeRounded",
    "node_id": "4229:32399",
  },
  {
    "name": "ReplayRounded",
    "node_id": "4229:31973",
  },
  {
    "name": "TimerRounded",
    "node_id": "4229:31970",
  },
  {
    "name": "OfflineBoltRounded",
    "node_id": "4229:31958",
  },
  {
    "name": "VisibilityOffRounded",
    "node_id": "4229:31996",
  },
  {
    "name": "VisibilityRounded",
    "node_id": "4229:31999",
  },
  {
    "name": "DragIndicatorRounded",
    "node_id": "4229:31948",
  },
  {
    "name": "LocalParkingRounded",
    "node_id": "4229:31960",
  },
  {
    "name": "Steppers",
    "node_id": "4689:52",
  },
]
`;
